src/anemoi/training/train/forecaster.py

```diff
@@ -217,6 +220,24 @@ def get_loss_function(
 
         return loss_function
 
+    def training_weights_for_imputed_variables(
+        self,
+        batch: torch.Tensor,
+    ) -> None:
+        """Update the loss weights mask for imputed variables."""
+        loss_weights_mask = torch.ones((1, 1), device=batch.device)
+        # iterate over all pre-processors and check if they have a loss_mask_training attribute
+        for pre_processor in self.model.pre_processors.processors.values():
+            if hasattr(pre_processor, "loss_mask_training"):
+                loss_weights_mask = loss_weights_mask * pre_processor.loss_mask_training
+            # if transform_loss_mask function exists for preprocessor apply it
+            if hasattr(pre_processor, "transform_loss_mask"):
+                loss_weights_mask = pre_processor.transform_loss_mask(loss_weights_mask)
+        # update scaler with loss_weights_mask retrieved from preprocessors
+        if "loss_weights_mask" in self.loss.scalar:
```

Comment:
Would it not make sense to have this if at the start so that the above multiplications won't unneccesarily take place.

Reply:
@sahahner Will this be addressed?